test-suite-data/PURL-TYPES.maven.json

```diff
@@ -0,0 +1,121 @@
+[
+  {
+    "description": "valid maven purl",
+    "purl": "pkg:maven/org.apache.commons/io@1.3.4",
+    "canonical_purl": "pkg:maven/org.apache.commons/io@1.3.4",
+    "type": "maven",
+    "namespace": "org.apache.commons",
+    "name": "io",
+    "version": "1.3.4",
+    "qualifiers": null,
+    "subpath": null,
+    "is_invalid": false
+  },
+  {
+    "description": "basic valid maven purl without version",
+    "purl": "pkg:maven/org.apache.commons/io",
+    "canonical_purl": "pkg:maven/org.apache.commons/io",
+    "type": "maven",
+    "namespace": "org.apache.commons",
+    "name": "io",
+    "version": null,
+    "qualifiers": null,
+    "subpath": null,
+    "is_invalid": false
+  },
+  {
+    "description": "maven often uses qualifiers",
+    "purl": "pkg:Maven/org.apache.xmlgraphics/batik-anim@1.9.1?classifier=sources&repositorY_url=repo.spring.io/release",
```

Comment:
Isn't this an **invalid** example, since the solidus `/` in `repo.spring.io/release` is not percent encoded?

Reply:
No. The parser is supposed to split on `?` before `/`.

Reply:
What about #439? I assume that `/` must be unencoded, when used as a PURL separator and encoded otherwise?